### PR TITLE
Apply "Send/Receive: Add BIP21 support" changes

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR applies breaking changes from
- https://github.com/breez/breez-sdk-liquid/pull/414

Full integration will be in scope of another PR:
- https://github.com/breez/misty-breez/issues/156

### Breaking changes:
* `PrepareReceivePaymentResponse`, `PrepareReceiveOnchainResponse` consolidated into `PrepareReceiveResponse`
* `ReceiveOnchainResponse` consolidated into `ReceivePaymentResponse`
* `PrepareSendRequest` now has `destination` & an optional `amountSat`
* `prepareRes` fields on structs got renamed to `prepareResponse`
* `PrepareReceiveResponse` now has `paymentMethod`

### Changelist:
- Apply "Send/Receive: Add BIP21 support" changes 
- Log `PaymentData` itself on SDK event listener  4334f6a
- Remove obsolete matchers on payment title e9b55ad
- Update `PaymentData` fields  e314fea
    - Added `toJson` & `fromJson` extensions on `PaymentDetails` 5d197737f2ff44f08fcf6e1943d638c7578908a2
- Create a `maybeMap` extension of `PaymentDetails` 380be17
- Reflect `PaymentData` changes and rely on `maybeMap` extension for matching values by details type 2fcc825
- Update payment tracking logic ee80686
- Get payment description from payment details 4106fcb
- Apply rename changes 92b08058c06432ba501aaa6dfac2aabd9e5f521e
  - `amountSat` -> `payerAmountSat`
  - `PaymentDetails_Liquid.address` -> `PaymentDetails_Liquid.destination`
- Update dependencies to latest & embed breez-sdk-liquid framework d77acdd8b17de78ffc6fe19a63a0c0010ef1e9c6

### Other changes:
* Re-enabled `description` on `ReceiveChainSwapPage`

### CODE REVIEW

**Review by ok300 — APPROVED** [review]

File: lib/routes/chainswap/receive/chainswap_qr_dialog.dart [hunk 1]
```diff
@@ -64,9 +64,9 @@ class ChainSwapQrDialogState extends State<ChainSwapQrDialog> with SingleTickerP
   @override
   void didUpdateWidget(covariant ChainSwapQrDialog oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (widget.receiveOnchainResponse?.address != oldWidget.receiveOnchainResponse?.address) {
+    if (widget.receivePaymentResponse?.destination != oldWidget.receivePaymentResponse?.destination) {
       var inputCubit = context.read<InputCubit>();
-      inputCubit.trackPayment(widget.receiveOnchainResponse!.bip21).then((value) {
+      inputCubit.trackPayment(widget.receivePaymentResponse!.destination).then((value) {
```
Comment (dangeross) [inline lib/routes/chainswap/receive/chainswap_qr_dialog.dart:69]: I think this needs checking. The destination is the BIP21 for a chain swap, but the `trackPayment` function is comparing it to the payment details swapId.

Comment (hydra-yse) [reply lib/routes/chainswap/receive/chainswap_qr_dialog.dart:69]: Yes, Ross is right. We should compare the destination to the `Payment.destination` instead.

**Review by hydra-yse — COMMENTED** [review]
Looking good! Just the `trackPayment` issue. @erdemyerebasmaz Do you think we need to better document this on the SDK side? Maybe it's just something that better belongs in the doc snippets.

**Review by hydra-yse — APPROVED** [review]
LGTM

**Review by dangeross — APPROVED** [review]

**Pull request merged**